### PR TITLE
Feature/state management

### PR DESCRIPTION
**Pull request by RobertPorterfield**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Update the version number in package.json when submitting your PR
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+
+## [Unreleased]
+- (Keep your changes here until you have a release version)
+
+## [0.1.0] - 2022-05-06
+### Added
+- This CHANGELOG file.
```
Comment (GregoryForrest) [inline CHANGELOG.md:13]: @RobertPorterfield -- Should you have some notes here?

File: src/index.tsx [hunk 2]
```diff
@@ -13,7 +13,8 @@ ReactDOM.render(
       integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3"
```
Comment (GregoryForrest) [inline src/index.tsx:13]: @RobertPorterfield  -- Do we want to link to something hosted on SharePoint vs a Global CDN?

File: src/stateManagement/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    case 'PURGE_STATE':
+      return initialState;
+    default:
+      return state;
+
+  }
+
+}
+
+export default Reducer;
```
Comment (GregoryForrest) [inline src/stateManagement/reducer.ts:37]: @RobertPorterfield  -- Haven't see this before --- Github doesn't like that you don't have a newline at the end of the file.
image

File: src/App.tsx [hunk 4]
```diff
@@ -7,55 +7,40 @@ import PizZipUtils from 'pizzip/utils/index.js';
...
-  const [options, setOptions] = useState<any>({
-    program_mod_system_name: "",
-    TMCRDate: date.toDateString(),
-    type_of_contract: "Firm Fixed Price",
-    attachment_number: null,
-    cdrl_sequence_number: null,
-    exhibit: null,
-    rfp_contract: null,
-    CLIN: null,
-    include_stuff: false
```
Comment (GregoryForrest) [inline src/App.tsx:26]: @RobertPorterfield  -- Assuming the elements that went away and did not get included in the Global Wizard state will be created as they are needed based on Sprints, and that these were just initially here for testing?

File: src/Steps/Steps.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { Step1 } from "./Step1";
+import { FinalStep } from "./FinalStep";
+
+export const TMCRStepNames = ["Home", "TO Info", "TO Program Requirements", "TMSS Requirements", "IETM Functionaly Requirements", "Mandatory Requirements", "Configurable Requirements", "Delivery Requirements", "Spec/Std Interface Records (SIRS)", "Attachment 1", "Attachment 2", "Review/Print"];
+export const TMCRFinalStep = 11;
```
Comment (GregoryForrest) [inline src/Steps/Steps.tsx:5]: @RobertPorterfield  - Should we base the TMCRFinalStep on the number of elements in the WizardSteps?  That way if the array changes (up or down) we don't have to worry about forgetting to update the max number.  Same with the TMCRStepNames -- would it be good to merge that into an array of Names/ReactObj?

File: src/stateManagement/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        ...state, wizardStep: state.wizardStep - 1
+      }
+    case 'GOTO_STEP':
+      return {
+        //...state, wizardStep: 1
+        ...state, ...action.payload
+      }
+    case 'PURGE_STATE':
+      return initialState;
+    case 'ADD_TMCR':
```
Comment (GregoryForrest) [inline src/stateManagement/reducer.ts:28]: Assuming this is a placeholder for future capability -- as right now it just returns the same state.

File: src/App.tsx (hunk 4 above)
Comment (RobertPorterfield) [reply src/App.tsx:26]: You got it! This was placeholder/demo stuff

File: CHANGELOG.md (hunk 1 above)
Comment (RobertPorterfield) [reply CHANGELOG.md:13]: Yes I absolutely should! 😆 
I'll add a summary

File: src/index.tsx (hunk 2 above)
Comment (RobertPorterfield) [reply src/index.tsx:13]: Good question. I'm conflicted on it to be honest. On the one hand we have control over everything in SharePoint, and it guarantees if they can get to our page that it'll work. That's the track I've traditionally followed. On the other hand, caching... that integrity attribute (holds hash value of the linked file) gives us some measure of confidence that what's being pulled from the CDN is what we're looking for. Do you have strong feelings one way or the other?

File: src/Steps/Steps.tsx (hunk 5 above)
Comment (RobertPorterfield) [reply src/Steps/Steps.tsx:5]: Agree. Current implementation is ugly. I'll add a small fix and a TODO to merge the steps and names into a single object. Next PR will define all the major steps and build out several of them., I'll include changes with that.

File: src/index.tsx (hunk 2 above)
Comment (RobertPorterfield) [reply src/index.tsx:13]: Issue #4

**Review by GregoryForrest — APPROVED** [review]
Reviewed and discussed

**Comment by ChristopherBlake8** [comment]
Good afternoon Rob and Greg, 
So we have one more to pull, if not mistaken?  I think there was four programs, right?

I just meet with Mr. Echols for 30 minutes and it went well.  Thanks so much!


As always it has been a pleasure, and please don’t hesitate to reach out if you need anything.  

Have a Marvelous Day!

MS. Christopher Blake Heath
Business Process Automation Specialist
Capabilities & Integration Branch
Contractor Supporting the Mission from Credence Management Solutions LLC
AFLCMC/OZIC at Wright-Patterson AFB, OH 
Cell (423) 366-9212
***@***.***
**Working Remotely

-----Original Message-----
From: GregoryForrest ***@***.***> 
Sent: Monday, May 23, 2022 12:55 PM
To: OZI-SP-Dev/tmcr ***@***.***>
Cc: HEATH, CHRISTOPHER B CTR USAF AFMC AFLCMC/XP-OZ/OZI ***@***.***>; Review requested ***@***.***>
Subject: [URL Verdict: Neutral][Non-DoD Source] Re: [OZI-SP-Dev/tmcr] Feature/state management (PR #3)

Merged #3 <https://github.com/OZI-SP-Dev/tmcr/pull/3>  into main.

**Pull request merged**